var/spack/repos/builtin/packages/gperftools/package.py

```diff
@@ -22,4 +22,22 @@ class Gperftools(AutotoolsPackage):
     version('2.4', sha256='982a37226eb42f40714e26b8076815d5ea677a422fb52ff8bfca3704d9c30a2d')
     version('2.3', sha256='093452ad45d639093c144b4ec732a3417e8ee1f3744f2b0f8d45c996223385ce')
 
-    depends_on("unwind")
+    variant('sized_delete', default=False, description="Build sized delete operator")
+    variant('dynamic_sized_delete_support', default=False, description="Try to build run-time switch for sized delete operator")
+    variant('debugalloc', default=True, description="Build versions of libs with debugalloc")
+    variant('libunwind', default=True, description="Enable libunwind linking")
+
+    depends_on("unwind", when="+libunwind")
+
+    def configure_args(self):
+        args = []
+        for opt in ('sized_delete', 'dynamic_sized_delete_support'):
+            if '+{0}'.format(opt) in self.spec:
+                args.append('--enable-{0}'.format(opt.replace('_', '-')))
+            else:
+                args.append('--disable-{0}'.format(opt.replace('_', '-')))
```

Comment:
To simplify (reducing duplication of the opt substitution):
```suggestion
            args.append('--{0}-{1}'.format('enable' if self.spec.variants[opt].value else 'disable', opt.replace('_', '-')))
```